core/src/test/java/org/apache/spark/unsafe/map/AbstractBytesToBytesMapSuite.java

```diff
@@ -667,4 +668,53 @@ public void testPeakMemoryUsed() {
     }
   }
 
+  @Test
+  public void avoidDeadlock() throws InterruptedException {
```

Comment:
Hi, @viirya . Since this test case reproduces `Deadlock` situation, we need a timeout logic. Otherwise, it will hang (instead of failures) when we hit this issue later.

Reply:
I've tried few ways to set a timeout logic, but don't work. The deadlock hangs the test thread.